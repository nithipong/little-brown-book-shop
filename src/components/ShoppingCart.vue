<template>
  <div class="shopping-cart-container">
    <h2 class="title is-5">Shopping Cart</h2>
    <section v-if="shoppingCart.count > 0">
      <table class="table shopping-cart-summary is-fullwidth is-hoverable">
        <tbody>
          <tr v-for="product in shoppingCart.products"
            :key="product.id">
            <td width="100px">
              <figure class="image">
                <img :src="product.cover"
                  :alt="product.title">
              </figure>
            </td>
            <td>
              <h3 class="is-size-5 has-text-weight-bold">{{ product.title }}</h3>
            </td>
            <td class="column-qty is-size-5">x{{ product.qty }}</td>
            <td class="column-price is-size-5">
              {{ (product.price * product.qty) | formatCurrency }}
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr v-if="shoppingCart.discount">
            <td colspan="3"
              class="has-text-right">
              Discount <span class="">({{ shoppingCart.discountText }})</span>
            </td>
            <td class="has-text-right summary-discount">
              {{ shoppingCart.discount | formatCurrency }}
            </td>
          </tr>
          <tr>
            <td colspan="3"
              class="has-text-right has-text-weight-bold">
              Net
            </td>
            <td class="has-text-right has-text-weight-bold summary-net">
              {{ shoppingCart.net | formatCurrency }}
            </td>
          </tr>
        </tfoot>
      </table>
    </section>
    <section v-else>
      Cart is Empty
    </section>
  </div>
</template>

<script>
import { mapActions } from "vuex";
export default {
  props: {
    "props-shopping-cart": Object
  },
  data() {
    return {
      shoppingCart: Object.assign({}, this.propsShoppingCart)
    };
  },
  methods: {
    ...mapActions(["addToShoppingCart", "updateProductQty"])
  }
};
</script>

<style lang="sass">
.shopping-cart-container
  background-color: white
  padding: 15px
  .shopping-cart-summary
    margin-bottom: 10px
    .column-qty
      width: 30px
      text-align: right
    .column-price
      width: 50px
      text-align: right
</style>
