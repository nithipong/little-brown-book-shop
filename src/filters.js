import Vue from 'vue';
import numeral from 'numeraljs';

Vue.filter('formatCurrency', value => {
  if (value) return numeral(value).format('0,0[.]00');
  return value;
});
