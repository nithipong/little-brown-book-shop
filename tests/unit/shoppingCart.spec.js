import { shallowMount } from '@vue/test-utils';
import ShoppingCart from '@/components/ShoppingCart.vue';
// filter
import '@/filters.js';

describe('ShoppingCart', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(ShoppingCart);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("renders correctlye", () => {
    const wrapper = shallowMount(ShoppingCart);
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.text()).toContain('Cart is Empty');
    expect(wrapper.text()).toContain('Shopping Cart');
  });

  it("render summary", () => {
    const wrapper = shallowMount(ShoppingCart, {
      propsData: {
        propsShoppingCart: {
          products: [
            {
              "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855652.jpg",
              "price": "350",
              "title": "Harry Potter and the Philosopher's Stone (I)",
              "id": "9781408855652"
            },
            {
              "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855669.jpg",
              "price": "350",
              "title": "Harry Potter and the Chamber of Secrets (II)",
              "id": "9781408855669"
            }
          ],
          subtotal: 0,
          discount: 70,
          discountText: "10%",
          net: 630,
          count: 2
        }
      }
    });
    expect(wrapper.find('.summary-discount').text()).toContain(70);
    expect(wrapper.find('.summary-net').text()).toContain(630);
  });



});
