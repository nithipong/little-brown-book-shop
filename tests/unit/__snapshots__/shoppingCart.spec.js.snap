// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCart renders correctlye 1`] = `
<div
  class="shopping-cart-container"
>
  <h2
    class="title is-5"
  >
    Shopping Cart
  </h2>
   
  <section>
    
    Cart is Empty
  
  </section>
</div>
`;
